### PR TITLE
[STORM-949] On the topology summary UI page, added Elapsed time since error column.

### PR DESCRIPTION
**Pull request by jerrypeng**
Current on the topology summary UI page and tells you the last error and highlights it red if it happened within the last 30 min.  However, I think its useful if there was a column that told you the time that has elapsed since the most recent error occurred.  I found this useful for monitoring the well being of my storm cluster.

### CODE REVIEW

**Comment by jerrypeng** [comment]
Not sure why travis is failing again.  I just built and ran all tests on my local machine and everything was fine

**Comment by HeartSaVioR** [comment]
Hi, there's a storm-hive compilation issue.
Your patch modifies html pages, so you don't need to worry about build. :)
I'll take a look when I have some time. Thanks!

File: storm-core/src/ui/public/templates/topology-page-template.html [hunk 1]
```diff
@@ -313,6 +320,10 @@
         </th>
         <th class="header">Last error
         </th>
+		<th class="header">
```
Comment (HeartSaVioR) [inline storm-core/src/ui/public/templates/topology-page-template.html]: MINOR: Indentation goes wrong.

File: storm-core/src/ui/public/templates/topology-page-template.html [hunk 2]
```diff
@@ -244,6 +248,9 @@
         <td>
           <span id="{{errorLapsedSecs}}" class="errorSpan">{{lastError}}</span>
         </td>
+	 	<td>
```
Comment (HeartSaVioR) [inline storm-core/src/ui/public/templates/topology-page-template.html]: MINOR: Indentation goes wrong.

File: storm-core/src/ui/public/templates/topology-page-template.html [hunk 3]
```diff
@@ -226,6 +226,10 @@
         </th>
         <th class="header">Last error
         </th>
+		<th class="header">
```
Comment (HeartSaVioR) [inline storm-core/src/ui/public/templates/topology-page-template.html]: MINOR: Indentation goes wrong.

File: storm-core/src/ui/public/templates/topology-page-template.html [hunk 4]
```diff
@@ -333,6 +344,9 @@
         <td>
           <span id="{{errorLapsedSecs}}" class="errorSpan">{{lastError}}</span>
         </td>
+		<td>
```
Comment (HeartSaVioR) [inline storm-core/src/ui/public/templates/topology-page-template.html]: MINOR: Indentation goes wrong.

**Comment by HeartSaVioR** [comment]
@jerrypeng 
I'd like to see applied screenshot first before taking a detail look. Could you post it to comment?

File: storm-core/src/ui/public/templates/topology-page-template.html (hunk 1 above)
Comment (caofangkun) [inline storm-core/src/ui/public/templates/topology-page-template.html]: @HeartSaVioR 
Code style shoud be checked before commit , shall we provide ```.editorconfig ``` with ````git pre-commit hooks```  ?

Comment (HeartSaVioR) [inline storm-core/src/ui/public/templates/topology-page-template.html]: @caofangkun 
Before that we need to establish code style for Java, HTML, etc. And have formatter file to let users applies it easily.

For example, Jedis uses Eclipse Java formatter which HBase project is using.
https://github.com/xetorthio/jedis/blob/master/hbase-formatter.xml
And ```make format``` formats whole files, so any contributors can run that commands before commit.

Comment (jerrypeng) [inline storm-core/src/ui/public/templates/topology-page-template.html]: +1

**Comment by jerrypeng** [comment]
screen shot 2015-08-12 at 9 45 59 am

**Comment by jerrypeng** [comment]
fixed the formating issues

**Comment by knusbaum** [comment]
I'm more in favor of putting a time/date here than an elapsed time.

The actual description for STORM-949 is "On the topology summary UI page, last shown error should have the time and date".

**Comment by jerrypeng** [comment]
Yes but the exact time and date can be found if you drill down to the component.  Having the elapsed time since time I feel like will tell administrators in a finer grain when the error happened

**Comment by jerrypeng** [comment]
@HeartSaVioR can you take a look at my pull request again

**Comment by HeartSaVioR** [comment]
@jerrypeng 
I'm with @knusbaum. PR is a bit different from JIRA title, and I'm also more in favor of putting time/date.
How about gathering consensus about this feature from dev mailing list and reflect feedback?

**Comment by jerrypeng** [comment]
Just modified the UI to have error time shown as a time and date.  If you hover over that time a tooltip will pop up displaying the elapsed time.
new_ui

**Comment by HeartSaVioR** [comment]
Modified version of UI is what I and @knusbaum , @d2r suggested, and it seems that there're no other opinions now.
Build failure is not related to.
So I'm +1.

**Pull request merged**